File: week3/Eren-Taskiran/19-remove-nth-node-from-end-of-list.java
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
       ListNode ans = new ListNode(0);
        ans.next = head;
        
        int length = 0;
        ListNode tmp = head;
        while (tmp != null) {
            length++;
            tmp = tmp.next;
        }
        
        length -= n;
        tmp = ans;
        while (length > 0) {
            length--;
            tmp = tmp.next;
        }
        
        tmp.next = tmp.next.next;
        
        return ans.next;
    }
}

File: week3/Can-Yavuz/19-remove-nth-node-from-end-of-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode RemoveNthFromEnd(ListNode head, int n) {
        ListNode dummyHead = new ListNode(0);
        dummyHead.next = head;

        ListNode first = dummyHead, second = dummyHead; 

        for (int i = 0; i <= n; i++) {
            second = second.next;
        }
        while (second != null) {
            first = first.next;
            second = second.next;
        }
        first.next = first.next.next;
        
        return dummyHead.next;
    }
}